### DISCUSSION TITLE
Compile tgfx as a independent library.

### DISCUSSION THREAD

**Discussion by peerless2012** [Tencent/libpag#1811] [Ideas] [answered]
As the tgfx code is move to a seperate git repo, and make tgfx a independent library, not only support for pag.

So it's time to compile tgfx as a independent library (dll、so、etc)，and other customer function can use tgfx directly。

**Comment by domchen** [comment, marked as answer]
We hide all of the tgfx APIS in the PAG project for a reason. Similar to Skia, tgfx is not designed to maintain binary compatibility due to its large set of APIs that are subject to change over time.  We strongly advise every app or framework to embedding its own tgfx and keep its APIs hidden. So we only support building the static library for tgfx for now.  In the future, as the APIs of tgfx become more stable, we might consider allowing its APIs to be exposed across various apps and frameworks.
